main/src/ca/uwaterloo/flix/language/phase/jvm/ClassMaker.scala

```diff
@@ -32,20 +32,20 @@ class ClassMaker(visitor: ClassWriter) {
     makeField(fieldName, fieldType, v, f, i)
   }
 
-  def mkConstructor(f: BytecodeInstructions.Instruction, descriptor: MethodDescriptor, v: Visibility): Unit = {
-    mkMethod(f, JvmName.ConstructorMethod, descriptor, v, Instanced)
+  def mkConstructor(f: BytecodeInstructions.InstructionSet, descriptor: MethodDescriptor, v: Visibility): Unit = {
+    mkMethod(f, JvmName.ConstructorMethod, descriptor, v, Implementable, Instanced)
   }
 
-  def mkStaticConstructor(f: BytecodeInstructions.Instruction): Unit =
-    mkMethod(f, JvmName.StaticConstructorMethod, MethodDescriptor.NothingToVoid, Default, Static)
+  def mkStaticConstructor(f: BytecodeInstructions.InstructionSet): Unit =
+    mkMethod(f, JvmName.StaticConstructorMethod, MethodDescriptor.NothingToVoid, Default, Implementable, Static)
 
-  private def mkMethod(f: BytecodeInstructions.Instruction, methodName: String, descriptor: MethodDescriptor, v: Visibility, i: Instancing): Unit = {
-    val modifier = ClassMaker.valueOf(v) + ClassMaker.valueOf(i)
-    val methodVisitor = visitor.visitMethod(modifier, methodName, descriptor.toString, null, null)
-    methodVisitor.visitCode()
-    f(new BytecodeInstructions.F(methodVisitor))
-    methodVisitor.visitMaxs(999, 999)
-    methodVisitor.visitEnd()
+  def mkMethod(ins: BytecodeInstructions.InstructionSet, methodName: String, descriptor: MethodDescriptor, v: Visibility, f: Finality, i: Instancing): Unit = {
+    val m = ClassMaker.valueOf(v) + ClassMaker.valueOf(f) + ClassMaker.valueOf(i)
```

Comment:
I think instead Visibility and so forth should have `toInt` or `valueOf` etc. Instead of having to call it on ClassMaker.